src/iosource/PktSrc.cc

```diff
@@ -385,6 +399,14 @@ void PktSrc::Process()
 			}
 		break;
 		}
+	default:
```

Comment:
The idea was to put the determination of the L2 and L3 protocols in one place rather than having it spread across several files. This combines the default handling provided by old line 275 above (pkt_hdr_size = props.hdr_size) with the fallback processing previously in NetSessions::NextPacket (old lines 202-228 in src/Sessions.cc). 

So, in the old code, everything would just fall through here with pkt_hdr_size set to whatever's appropriate for the link, then in NextPacket, we assume that what we're looking at is IP if we can recognize it. So, it's really no change in function.

Also, note that PcapSource is written generically. It handles link types other than those explicitly called out here. So it makes sense to just trust that GetLinkHeaderSize will get you to the layer 3 header and then try to process it as IP.

The l3_proto determination could probably be enhanced here by checking (ip->ip_v == 6) before assuming AF_INET6 (and otherwise setting it to AF_UNSPEC or something).